### ISSUE TITLE
Deduplicate email-recipients for new issue notifications

### ISSUE THREAD

**Issue by mittinatten** [pkp/pkp-lib#3460]
This is a feature request based on https://forum.pkp.sfu.ca/t/ojs-3-same-email-announcement-sent-twice-to-some-users/37603

Problem: Users with several roles will sometimes get several copies of an announcement email. The list of recipients needs to be deduplicated before sending.

**Comment by asmecher** [comment]
@mittinatten, is your account that received two messages listed in `notification_mail_list`? It's possible that you got a notification through the normal notification system and another through the mailing list feature -- I don't see at a glance where duplicates could be sent out otherwise.

**Comment by mittinatten** [comment]
Right, it was a new issue-notification, not an announcement per se. I think it was only editors that got two emails (most of them are reviewers and/or authors as well).

**Comment by asmecher** [comment]
@mittinatten, did you check if your account is listed in `notification_mail_list`?

**Comment by mittinatten** [comment]
There is only one user in that list. One of the users who got two emails has the following roles:

Name | designation
--|--
Reviewer | R
Section editor | SecE
Author | AU
Reader | Read
Journal editor | JE

The admin user (me) only got one email, and has the following roles

Name | Designation
-- | --
Journal manager | JM
Journal editor | JE

**Comment by asmecher** [comment]
@mittinatten, I'm having a hard time tracking this down, either through local testing or code examination. The code that fetches the user list is de-duplicated on the SQL side with a `SELECT DISTINCT`.

Are you sure there aren't e.g. two users in the journal that forward to the same email account? That still seems the most likely cause to me.

If not, I'd suggest adding some `error_log` calls to the email send code -- the `send` function in `lib/pkp/classes/mail/Mail.inc.php` -- to capture what addresses are being added in the recipient, CC, and BCC fields. This should capture the intended recipient list according to OJS, so we can determine whether there are actual duplicates there, or e.g. whether mail delivery is somehow adding something, etc.

**Comment by mittinatten** [comment]
Hi,
I added some debug messages as suggested and couldn't find any duplicate emails when I published a test issue.

I see that some editors have test accounts, i.e. one official account with their institutional email and another with gmail or similar. Maybe they got confused about this somehow (even though this was the first thing I asked when I got the error report).

Sorry for bothering about a non-issue. I will reopen if something pops up.

Simon
